src/checkout/checkout-service.ts

```diff
@@ -1014,22 +1014,23 @@ export default class CheckoutService {
     }
 
     /**
-     * Deletes a payment instrument by an id.
+     * Deletes a payment instrument by an id and currency.
      *
      * Once an instrument gets removed, it can no longer be retrieved using
      * `CheckoutStoreSelector#getInstruments`.
      *
      * ```js
-     * const state = service.deleteInstrument('123');
+     * const state = service.deleteInstrument('123', 'USD');
      *
      * console.log(state.data.getInstruments());
      * ```
      *
      * @param instrumentId - The identifier of the payment instrument to delete.
+     * @param currencyCode - The currency code as selected by the shopper.
      * @returns A promise that resolves to the current state.
      */
-    deleteInstrument(instrumentId: string): Promise<CheckoutSelectors> {
-        const action = this._instrumentActionCreator.deleteInstrument(instrumentId);
+    deleteInstrument(instrumentId: string, currencyCode: string): Promise<CheckoutSelectors> {
```

Comment:
Is the currency code strictly necessary to delete an instrument? If so, this seems like a breaking change/

Reply:
The future potential is we'll eventually be moving everything to require a currency code. I'm happy to keep it as optional for now though.

Reply:
@capsula4 I do have a question though:
```
const { code } = config.shopperCurrency;
```
If I make the currency optional, does that imply this can be undefined? Or am I going about it the wrong way?

Reply:
@PharaohAsh if we will eventually move things to require a currency code, that could break custom checkouts. If we plan to actually make currency code a requirement in the API is not enough by fixing the SDK as partners have to manually migrate to a newer version, or they could even be integrating against Bigpay API directly. We might need to communicate this and give them time to migrate. I would prefer we keep it optional. 

On the other hand `shopperCurrency` should always be defined. If the shopper hasn't selected manually any currency, it should match the default currency. Keep in mind that shopper currency includes display currencies which I guess are not eligible for a parameter for deleting an instrument.

Reply:
@capsula4  Ah, right. That makes sense. I didn't know about the custom checkouts... that's complex indeed.

In regards to display currencies vs transactional currencies: theoretically, we should not be storing an instrument against a display currency and hence there should not be a case where you will be listing instruments for that particular display currency. In turn you wouldn't get a case where you'd delete an instrument under a display currency either.

Which is actually a very good point you've raised. Is there a way to determine what is and isn't a transactional currency?

Reply:
@capsula4 It looks like I'll have to decide the currency type on BCApp and prevent users from adding cards on display currencies.